docs/hack_loki_operator.md

```diff
@@ -0,0 +1,280 @@
+# Hacking on Loki Operator
+
+Loki Operator is the Kubernetes Operator for [Loki](https://grafana.com/docs/loki/latest/) provided by the Red Hat OpenShift engineering team.
+
+## Hacking on Loki Operator using kind
+
+[kind](https://kind.sigs.k8s.io/docs/user/quick-start/) is a tool for running local Kubernetes clusters using Docker container "nodes". kind was primarily designed for testing Kubernetes itself, but may be used for local development or CI.
+
+### Requirements
+
+* Install [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) or [Openshift CLI](https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/getting-started-cli.html) for communicating with the cluster. The guide below will be using `kubectl` for the same.
+* Create a running Kubernetes cluster using kind.
+* A container registry that you and your Kubernetes cluster can reach. We recommend  [quay.io](https://quay.io/signin/).
+
+### Installation of Loki Operator
+
+* Build and push the container image and then deploy the operator with:
+
+  ```console
+  make oci-build oci-push deploy REGISTRY_ORG=$YOUR_QUAY_ORG VERSION=latest
+  ```
+
+  where `$YOUR_QUAY_ORG` is your personal [quay.io](http://quay.io/) account where you can push container images.
+
+  The above command will deploy the operator to your active Kubernetes cluster defined by your local [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/). The operator will be running in the `default` namespace.
+
+* You can confirm that the operator is up and running using:
+
+  ```console
+  kubectl get pods
+  ```
+  
+  You should see `controller-manager-xxxx` and `minio-xxxx` pods running.
+
+* Now you need to create a gateway secret [2] for the operator. This can be done using:
+
+  ```console
+  kubectl create secret generic test1 \
+    --from-literal=clientID="<CLIENT_ID>" \
+    --from-literal=clientSecret="<CLIENT_SECRET>" \
+    --from-literal=issuerCAPath="<ISSUER_CA_PATH>"
+  ```
```

Comment:
We can drop this as per now on kind we should use `lokistack_dev.yaml` that is not providing the gateway.